api/src/routes/edit.ts

```diff
@@ -1,10 +1,59 @@
 import express = require("express");
+import bodyParser = require("body-parser");
 import Config from "../models/Config";
-const router = express.Router();
+import Fedora from "../services/Fedora";
+import FedoraObjectFactory from "../services/FedoraObjectFactory";
+import MetadataExtractor from "../services/MetadataExtractor";
 import { requireToken } from "./auth";
 
+const router = express.Router();
+
 router.get("/models", requireToken, function (req, res) {
     res.json({ CollectionModels: Config.getInstance().collectionModels, DataModels: Config.getInstance().dataModels });
 });
 
+router.post("/object/new", requireToken, bodyParser.json(), async function (req, res) {
+    let parentPid = req?.body?.parent;
+    if (parentPid !== null && parentPid.length === 0) {
+        parentPid = null;
+    }
+    const noParent = (req.body.noParent ?? "0") === "1";
+
+    // Validate parent parameters:
+    if (noParent && parentPid !== null) {
+        res.status(400).send("Cannot set parent PID and no parent PID");
+        return;
+    }
+    if (!noParent && parentPid === null) {
+        res.status(400).send("Must set either parent or noParent");
+        return;
+    }
+    // Validate parent PID, if set:
+    if (parentPid !== null) {
+        const fedora = Fedora.getInstance();
+        const extractor = MetadataExtractor.getInstance();
+        const relsExt = await fedora.getDatastreamAsString(req.body.parent, "RELS-EXT");
```

Comment:
Is this parentPid?

Reply:
Is what parentPid? The purpose of this code is that if we have the ID of a parent object, we need to first validate that the parent object is eligible to contain a child before we make the attachment. We can retrieve that information from a datastream called RELS-EXT in the repository (since this is an RDF document containing the model types of the object).

Reply:
Oh, I understand your question now -- yes, that was erroneous and has now been corrected.